dom5.ts

```diff
@@ -312,20 +312,34 @@ export function treeMap<U>(node: Node, mapfn: (node: Node) => U[]): U[] {
   return results;
 }
 
+export type childNodesRetriever = (node: Node) => Node[] | undefined;
+
+const defaultChildNodes: childNodesRetriever = node => node.childNodes;
+
+export const childNodesIncludeTemplate: childNodesRetriever = node => {
+  if (node.nodeName === 'template')
+    return treeAdapters.default.getTemplateContent(node).childNodes;
+  return node.childNodes;
+};
+
 /**
  * Walk the tree down from `node`, applying the `predicate` function.
  * Return the first node that matches the given predicate.
  *
  * @returns `null` if no node matches, parse5 node object if a node matches.
  */
-export function nodeWalk(node: Node, predicate: Predicate): Node|null {
+export function nodeWalk(
+    node: Node,
+    predicate: Predicate,
+    retrieveChildNodes: childNodesRetriever = defaultChildNodes): Node|null {
```

Comment:
Interesting API addition. I'm slightly worried it might be a little overly general. What values would this have besides normal children and children with template contents?